src/arch/riscv/linux/fs_workload.cc

```diff
@@ -184,13 +184,37 @@ BootloaderKernelWorkload::loadKernel()
     }
 }
 
+void
+BootloaderKernelWorkload::loadInitrd()
+{
+    if (params().initrd_filename != "") {
+        inform("Loading initrd file: %s at address %#x\n",
+               params().initrd_filename, params().initrd_addr);
+
+        loader::ImageFileDataPtr initrd_file_data(
+            new loader::ImageFileData(params().initrd_filename));
```

Comment:
Initrd should not be decompressed https://github.com/gem5/gem5/pull/2569

```suggestion
        loader::ImageFileDataPtr initrd_file_data(
            new loader::ImageFileData(params().initrd_filename, false));
```